.github/workflows/generate-instance.yml

```diff
@@ -0,0 +1,55 @@
+name: Generate Cred Instance
+on:
+  # Trigger on merging to master.
+  push:
+    branches:
+      - master
+  # As well as every 6 hours.
+  schedule:
+    - cron: 0 */6 * * *
+
+jobs:
+  GenerateCredInstance:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false # Required to make github pages deployment work correctly
+
+      - name: Cache Data # Cache SourceCred Data, invalidating if any of the config changes or the SC version is updated
+        uses: actions/cache@v2
+        with:
+          path: '**/cache'
+          key: SC-${{ runner.os }}-${{ hashFiles('**/config.json', '**/sourcecred.json', '**/yarn.lock') }}
+
+      - name: Install Packages 🔧
+        run: yarn --frozen-lockfile
+
+      - name: Load Data into Graph 🔌
+        run: |
+          yarn load
+          yarn graph
+        env:
+          SOURCECRED_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SOURCECRED_DISCORD_TOKEN: ${{ secrets.SOURCECRED_DISCORD_TOKEN }}
+
+      - name: Compute Cred 🧮
+        run: yarn score
+
+      - name: Generate Frontend 🏗
+        run: |
+          yarn site
+          cp -rf ./output ./site
+          cp -rf ./data ./site
+          cp -rf ./config ./site
+          cp -f ./sourcecred.json ./site
+
+      - name: Copy CNAME if exists
+        run: '(test -f CNAME && cp ./CNAME ./site) || echo CNAME does not exist, skipping step'
```

Comment:
I like the idea of having this as a variable, rather than a file. Since if we put the CNAME in the master branch, it would be the only site deployment detail "leaking" into the cred instance.

Reply:
Although I guess this workflow leaks into the master branch too :)

I'm not sure if I can make non-secret environment variables in the UI, or if they all get committed to the branch too? I suppose checking them in makes more sense. I do think putting the CNAME into a checked-in value in the workflow would still be cleaner than leaking a CNAME file into the root.

Reply:
Just gonna get rid of it altogether. [According to GH docs](https://docs.github.com/en/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain), if the user sets a custom domain in the GitHub settings, then it will automatically put the CNAME file in the gh-pages branch. Tested it out on my repo: https://github.com/hammadj/example-instance/tree/gh-pages